functions/completeness.R

```diff
@@ -10,74 +8,82 @@ completeness <- function(end_date, level = c("board", "scotland")) {
+                                  "resource/03cf3cb7-41cc-4984-bff6-",
+                                  "bbccd5957679/download/quarters.csv")) %>%
     janitor::clean_names() %>%
+    dplyr::filter(is.na(completeness_qf)) %>%
+    dplyr::mutate(quarter = case_when(
+      str_detect(quarter, "Q1") ~ str_replace(quarter, "Q1", "Q4"),
+      str_detect(quarter, "Q2") ~ str_replace(quarter, "Q2", "Q1"),
+      str_detect(quarter, "Q3") ~ str_replace(quarter, "Q3", "Q2"),
+      str_detect(quarter, "Q4") ~ str_replace(quarter, "Q4", "Q3")
+    )) %>%
+    dplyr::mutate(quarter = 
+                    if_else(substr(quarter, 5, 6) == "Q4",
+                            paste0(parse_number(quarter) - 1, "Q4"),
+                            quarter)) %>%
     dplyr::filter(smr_type == "SMR01",
-                  readr::parse_number(financial_year) == 
+                  readr::parse_number(quarter) == 
                     lubridate::year(end_date) - 1) %>%
     dplyr::left_join(lookup, by = "hb2014") %>%
     dplyr::rename(board = hb2014name) %>%
     dplyr::mutate(board = replace(board,
                                   hb2014 == "S92000003",
                                   "Scotland")) %>%
-    tidyr::drop_na(board)
+    tidyr::drop_na(board) %>%
+    dplyr::select(board, quarter, completeness) %>%
+    tidyr::pivot_wider(names_from = quarter, values_from = completeness)
   
-  if (level == "board") {
-    
-    comp %<>%
-      dplyr::filter(board != "Scotland",
-                    completeness < 0.95) %>%
-      dplyr::mutate(completeness = paste0("(", 
-                                          scales::percent(completeness, 
-                                                          accuracy = 1),
-                                          ")")) %>%
-      tidyr::unite(var, board, completeness, sep = " ") %>%
-      dplyr::pull(var)
-    
-    if (length(comp) == 0) {
-      return(capture.output(
-        cat("All NHS board have SMR01 completeness of 95% and", 
-            "above for",
-            paste0(lubridate::year(end_date) - 1,
-                   "/",
-                   format(end_date, "%y")))))
-    } else {
-      return(capture.output(
-        cat("All NHS board have SMR01 completeness of 95% and",
-            "above for",
-            paste0(lubridate::year(end_date) - 1,
-                   "/",
-                   format(end_date, "%y")),
-            "with the exception of",
-            glue::glue_collapse(sort(comp), 
-                                sep = ", ", 
-                                last = " and "))))
-      
-    }
-    
-  }
+  fy <- readr::read_csv(paste0("https://www.opendata.nhs.scot/dataset/",
+                               "110c4981-bbcc-4dcb-b558-5230ffd92e81/",
+                               "resource/daf55fd2-457f-4845-9af1-5d154cc0b19c",
+                               "/download/financialyr.csv")) %>%
+    janitor::clean_names() %>%
+    dplyr::filter(smr_type == "SMR01",
```

Comment:
Shall we also filter is.na(completeness_qf) as qtr does? I know it doesn't matter as eventually you left joined fy to qtr, but just want to keep things tidy?